spec.html

```diff
@@ -37042,6 +37074,33 @@ <h1>
             <p>In case-insignificant matches when HasEitherUnicodeFlag(_rer_) is *false*, the mapping is based on Unicode Default Case Conversion algorithm toUppercase rather than toCasefold, which results in some subtle differences. For example, `Ω` (U+2126 OHM SIGN) is mapped by toUppercase to itself but by toCasefold to `ω` (U+03C9 GREEK SMALL LETTER OMEGA) along with `Ω` (U+03A9 GREEK CAPITAL LETTER OMEGA), so *"\u2126"* is matched by `/[ω]/ui` and `/[\u03A9]/ui` but not by `/[ω]/i` or `/[\u03A9]/i`. Also, no code point outside the Basic Latin block is mapped to a code point within it, so strings such as *"\u017F ſ"* and *"\u212A K"* are not matched by `/[a-z]/i`.</p>
           </emu-note>
         </emu-clause>
+
+        <emu-clause id="sec-updatemodifiers" type="abstract operation">
+          <h1>
+            UpdateModifiers (
+              _rer_: a RegExp Record,
+              _add_: a String,
+              _remove_: a String,
+            ): a RegExp Record
+          </h1>
+          <dl class="header">
+          </dl>
+          <emu-alg>
+            1. Let _ignoreCase_ be _rer_.[[IgnoreCase]].
+            1. Let _multiline_ be _rer_.[[Multiline]].
+            1. Let _dotAll_ be _rer_.[[DotAll]].
+            1. Let _unicode_ be _rer_.[[Unicode]].
+            1. Let _unicodeSets_ be _rer_.[[UnicodeSets]].
+            1. Let _capturingGroupsCount_ be _rer_.[[CapturingGroupsCount]].
+            1. If _add_ contains *"i"*, set _ignoreCase_ to *true*.
+            1. If _add_ contains *"m"*, set _multiline_ to *true*.
+            1. If _add_ contains *"s"*, set _dotAll_ to *true*.
+            1. If _remove_ contains *"i"*, set _ignoreCase_ to *false*.
+            1. If _remove_ contains *"m"*, set _multiline_ to *false*.
+            1. If _remove_ contains *"s"*, set _dotAll_ to *false*.
```

Comment:
```suggestion
            1. If _add_ contains *"i"*, set _ignoreCase_ to *true*.
            1. Else if _remove_ contains *"i"*, set _ignoreCase_ to *false*.
            1. If _add_ contains *"m"*, set _multiline_ to *true*.
            1. Else if _remove_ contains *"m"*, set _multiline_ to *false*.
            1. If _add_ contains *"s"*, set _dotAll_ to *true*.
            1. Else if _remove_ contains *"s"*, set _dotAll_ to *false*.
```

Use else-if to further emphasise that `add`/`remove` are disjoint.

Reply:
IMO, this doesn't indicate disjointedness any more than the independent Ifs do. Also, this suggestion prioritizes modifiers in _add_, while the existing text prioritizes modifiers in _remove_. Existing implementations in other languages that _don't_ error on `(?i-i:)`, such as C#/.NET, prioritize removal (i.e., set and then unset). While the consensus was to be more restrictive and issue an error, I'd still prefer we still maintain _remove_ priority in the specification text for the sake of consistency.

Reply:
If you would still prefer the "Else if", I would prefer to rewrite it as follows:

```suggestion
            1. If _remove_ contains *"i"*, set _ignoreCase_ to *false*.
            1. Else if _add_ contains *"i"*, set _ignoreCase_ to *true*.
            1. If _remove_ contains *"m"*, set _multiline_ to *false*.
            1. Else if _add_ contains *"m"*, set _multiline_ to *true*.
            1. If _remove_ contains *"s"*, set _dotAll_ to *false*.
            1. Else if _add_ contains *"s"*, set _dotAll_ to *true*.
```

Such that _remove_ continues to take precedence.

Reply:
That's fine too.

Reply:
@rbuckton I would still like you to do this.